guidelines/COMMIT_STYLE_GUIDE.md

```diff
@@ -0,0 +1,134 @@
+# Commit message format
+
+All commits on `main` branches must follow a format for the commit message (subject, body and optional footers)
+according to the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification. The basic
+convention is explained and extended in the following section.
+
+## Commit type
+
+We define the following types as supported by
+[commitlint/config-conventional](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional#type-enum),
+which in turn are derived from [Angular](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#type):
+
+- `build:` change the build system or external dependencies (for example, CMakeLists.txt, Dockerfile, package.xml)
+- `ci:` change CI configuration files and scripts (for example, GitHub actions and workflows)
+- `chore:` perform any general maintenance or work that is not accurately described by any other type
+- `docs:` change only documentation lines or files
+- `feat:` introduce, enhance or modify a feature
+- `fix:` resolve an issue, bug, or error
+- `perf:` improve performance without changing existing features or functionality
+- `refactor:` change code structure or organization without changing existing features or functionality
+- `release:` change only the package version and changelog to mark a new release
+- `revert:` rollback a previous commit without making any other changes
+- `style:` format / reformat code to adhere to style guidelines
+- `test:` introduce, enhance or modify unit tests
+
+## Commit scope
+
+The commit scope is optional and can be used to categorize the change. It should preferably be a single noun but may be
+a short phrase. For example,
+
+- `feat(api): <description>`
+- `fix(graph): <description>`
+- `docs(component services): <description>`
+
+For multi-package repositories, if a commit only changes a single package, the package name can be used as the commit
+scope. For example, `feat(base-components)` would indicate a feature in the `base_components` package.
+
+## Breaking changes
+
+If a commit includes breaking changes to the ABI or API of the package output, this must be clearly marked by adding an
+exclamation point `!` after the commit type (or after the scope, if there is one). In addition, the commit body should
+include a `BREAKING CHANGE:` description to clearly summarize the effects of the change.
+
+## Formatting
+
+Some additional formatting guidelines should also be respected:
+
+- The subject should be a single lowercase line, ending with the respective PR number in parentheses
```

Comment:
I may have missed that in the doc but this is quite tedious for commits no? Having the PR number in the changelog, yes but in the commit itself? That would mean you can't set your PR title before opening the PR because you don't know which number its gonna have

Reply:
Well these rules apply for commits to main, ergo after the PR is squashed. GitHub puts the PR number there automatically.

> All commits on `main` branches [...]

I did also write in the PR template

> Give the PR a relevant and descriptive title, using one of the supported commit types

But that doesn't mean that the PR title needs to explicitly refer to itself by number, or that the first commit on the feature branch should need to preemptively assume the PR number.

Let me know if there's a change that can make this clearer

Reply:
Okay yeah, I consider the commit style to be valid for all commits but it's true that we only want to enforce them for commits on main and then GitHub puts the number there automatically. All good